network/src/main/java/info/metadude/android/eventfahrplan/network/fetching/FetchFahrplan.kt

```diff
@@ -0,0 +1,190 @@
+package info.metadude.android.eventfahrplan.network.fetching
+
+import android.net.Uri
+import android.os.AsyncTask
+import android.os.Build
+
+import java.io.IOException
+import java.net.SocketTimeoutException
+import java.net.UnknownHostException
+import java.net.UnknownServiceException
+
+import javax.net.ssl.SSLException
+import javax.net.ssl.SSLHandshakeException
+
+import info.metadude.android.eventfahrplan.commons.logging.Logging
+
+import okhttp3.OkHttpClient
+import okhttp3.Request
+
+internal class FetchFahrplan(private val logging: Logging) {
+
+    private lateinit var task: FetchFahrplanTask
+    private lateinit var onFetchScheduleResult: (fetchScheduleResult: FetchScheduleResult) -> Unit
+    fun fetch(okHttpClient: OkHttpClient, url: String, eTag: String) {
+        task = FetchFahrplanTask(okHttpClient, logging, onFetchScheduleResult)
+        task.execute(url, eTag)
+    }
+
+    fun cancel() {
+        if (::task.isInitialized) {
+            task.cancel(true)
+        }
+    }
+
+    fun setListener(listener: (fetchScheduleResult: FetchScheduleResult) -> Unit) {
+        onFetchScheduleResult = listener
+        if (::task.isInitialized) {
+            task.setListener(listener)
+        }
+    }
+}
+
+class FetchFahrplanTask(
```

Comment:
Did you remove the `internal` intentionally?

Reply:
Nope, I'm just a bit Stupid :sweat_smile: